### PR TITLE
Add a function aligning numbers and a function for yielding for many cycles

### PR DESCRIPTION
**Pull request by xThaid**
Nothing controversial (I hope). I use these function in PR #258 and #271 , but as they aren't really related to the instruction cache, here they are.

### CODE REVIEW

File: coreblocks/utils/comb.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from amaranth import *
+
+
+def rotate_left(sig: Value, shift: int = 1) -> Value:
```
Comment (lekcyjna123) [inline coreblocks/utils/comb.py:4]: These two functions are already by default in amaranth (as a method in Value class)

File: test/utils/test_comb.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+from amaranth import *
+from amaranth.sim import *
+
+from coreblocks.utils import rotate_left, rotate_right
+
+from test.common import TestCaseWithSimulator
+
+
+class CircularShiftCircuit(Elaboratable):
```
Comment (lekcyjna123) [inline test/utils/test_comb.py:9]: I am not sure if this shouldn't be moved to coreblocks/utils.

Comment (tilk) [reply test/utils/test_comb.py:9]: Isn't this basically a test circuit? I don't see a practical need for generating all possible rotations at once.

Comment (xThaid) [reply test/utils/test_comb.py:9]: It was just a test circuit

**Review by tilk — APPROVED** [review]

**Review by lekcyjna123 — APPROVED** [review]

**Pull request merged**